vault_core/src/vault.rs

```diff
@@ -528,6 +603,46 @@ impl Vault {
         Ok(())
     }
 
+    /// Checks if a withdrawal is allowed based on the current vault limits.
+    ///
+    /// If both `epoch_snapshot_supported_token_amount` and `last_full_state_update_slot` are zero, this indicates that the vault has not yet recorded any withdrawals or state updates for the current epoch.
+    /// In this scenario, any withdrawal amount up to the total `max_withdrawable` tokens is allowed because the vault is effectively in an initial state.
+    ///
+    /// # Returns
+    /// * `Result<(), VaultError>` - Returns `Ok(())` if the withdrawal is within the allowed limit.
+    ///
+    /// # Errors
+    /// This function can return the following errors:
+    /// * [`jito_vault_sdk::error::VaultError::VaultOverflow`] - If any arithmetic operation (addition or multiplication) results in an overflow.
+    /// * [`jito_vault_sdk::error::VaultError::VaultWithdrawalLimitExceeded`] - If the requested withdrawal exceeds the allowed limit for the epoch.
+    pub fn check_withdrawal_allowed(&self, amount_to_withdraw: u64) -> Result<(), VaultError> {
+        let epoch_withdraw_amount: u128 = self.epoch_withdraw_supported_token_amount().into();
+        let epoch_snapshot_amount: u128 = self.epoch_snapshot_supported_token_amount().into();
+        let last_full_state_update_slot: u128 = self.last_full_state_update_slot().into();
+        let max_withdrawable = self.tokens_deposited();
+
+        if epoch_snapshot_amount == 0
```

Comment:
Why have this check? If nothing is in the vault, or nothing has been updated I don't think they should be able to withdraw anything

Reply:
Sorry, I was quite stupid...
You are totally right. I removed the check.